### PR TITLE
feat: Do not add views to retention_member_properties with updates

### PR DESCRIPTION
**Pull request by Tomvbe**
https://github.com/Informatievlaanderen/VSDS-LDESServer4J/issues/1062

### CODE REVIEW

File: ldes-server-infra-mongo/mongo-retention-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/MemberPropertiesRepositoryImpl.java [hunk 1]
```diff
@@ -66,12 +60,14 @@ public Optional<MemberProperties> retrieve(String id) {
...
 
 	@Override
-	public synchronized void addViewReference(String id, String viewName) {
-		Query query = new Query();
-		query.addCriteria(Criteria.where(ID).is(id));
-		Update update = new Update();
-		update.addToSet(VIEWS, viewName);
-		mongoTemplate.upsert(query, update, MemberPropertiesEntity.class);
+	public void addViewToAll(ViewName viewName) {
+		final var query = new Query();
```
Comment (jobulcke) [inline ldes-server-infra-mongo/mongo-retention-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/MemberPropertiesRepositoryImpl.java:64]: Shouldn't the view not only be added to the members that are in the collection whereto the view is added?

File: ldes-server-retention/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/integrationtest/stub/InMemoryMemberPropertiesRepository.java [hunk 2]
```diff
@@ -23,12 +23,7 @@ public InMemoryMemberPropertiesRepository() {
...
 
     @Override
-    public void saveMemberPropertiesWithoutViews(MemberProperties memberProperties) {
-        memberPropertiesMap.put(memberProperties.getId(), memberProperties);
-    }
-
-    @Override
-    public void save(MemberProperties memberProperties) {
+    public void insert(MemberProperties memberProperties) {
         memberPropertiesMap.put(memberProperties.getId(), memberProperties);
```
Comment (jobulcke) [inline ldes-server-retention/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/integrationtest/stub/InMemoryMemberPropertiesRepository.java:27]: As inserting only save the memberProperties when they does not exist, I suggest the use `putIfAbsent`

**Review by jobulcke — COMMENTED** [review]
see comments

File: ldes-server-infra-mongo/mongo-retention-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/MemberPropertiesRepositoryImpl.java (hunk 1 above)
Comment (Tomvbe) [reply ldes-server-infra-mongo/mongo-retention-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/MemberPropertiesRepositoryImpl.java:64]: This was indeed a bug..

File: ldes-server-retention/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/integrationtest/stub/InMemoryMemberPropertiesRepository.java (hunk 2 above)
Comment (Tomvbe) [reply ldes-server-retention/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/retention/integrationtest/stub/InMemoryMemberPropertiesRepository.java:30]: The database throws an exception if you call this twice so I adjusted the logic to throw an exception as well, when this gets called twice with the same record

**Review by jobulcke — APPROVED** [review]

**Pull request merged**